Task11/html5form

OleksiyRudenko opened the pull request:
Have imported the task from its dedicated repo.
Please, review.
[Live demo](https://kottans-frontend-2018.github.io/assignments_OleksiyRudenko/task_11/index.html)
[Source](https://github.com/kottans-frontend-2018/assignments_OleksiyRudenko/tree/task11/html5form/task_11)

OleksiyRudenko committed:
Initial commit

OleksiyRudenko committed:
Add reference to the font in use

OleksiyRudenko committed:
Add mock-ups

OleksiyRudenko committed:
Add mock-up author's notes

OleksiyRudenko committed:
Initialize index.html

OleksiyRudenko committed:
Add time tracking section

OleksiyRudenko committed:
Fix md formatting

OleksiyRudenko committed:
Add TOC

OleksiyRudenko committed:
Improve HTML layout

OleksiyRudenko committed:
Add web app logo

OleksiyRudenko committed:
Stylize logo

OleksiyRudenko committed:
Fix time tracking records

OleksiyRudenko committed:
Add html5 input validation

OleksiyRudenko committed:
html form: employ <fieldset>

OleksiyRudenko committed:
html form: Improve structure with html5 tags

OleksiyRudenko committed:
html form: Improve input browser hints

OleksiyRudenko committed:
Add monochromatic logo

OleksiyRudenko committed:
style: background colors, logo colouring

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
general: Attach stylesheet switch to logos onClick

OleksiyRudenko committed:
docs: Add related resources reference section

OleksiyRudenko committed:
style: Split styles between colours and dimensions

OleksiyRudenko committed:
mockup: Add a mockup with a main visual part 768px wide

OleksiyRudenko committed:
style: General styling

OleksiyRudenko committed:
style: Add text font

OleksiyRudenko committed:
style: Add icons font

OleksiyRudenko committed:
docs: Add remark on Proxima Nova font set error

OleksiyRudenko committed:
misc: Fix styling switch

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
mockup: Decomposition

OleksiyRudenko committed:
mockup: Decomposition (color)

OleksiyRudenko committed:
html: Improve layout, add elements

OleksiyRudenko committed:
style: Add colours

OleksiyRudenko committed:
style: Improve readability for icon only button

OleksiyRudenko committed:
style: Restyle checkbox (failed)

OleksiyRudenko committed:
style: Fix footer bg-color

OleksiyRudenko committed:
docs: Add references to styling radio buttons

OleksiyRudenko committed:
style: Customize radio-button

OleksiyRudenko committed:
style: Minor improvements

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: stylize radio button filedsets for PaymentMethod, ToneOfVoice

OleksiyRudenko committed:
docs: Add link to live page

OleksiyRudenko committed:
docs: Add link to live page and unstyling hint

OleksiyRudenko committed:
style: Improve fontawesome icons styling

OleksiyRudenko committed:
html: Switch from ul to table for order summary

OleksiyRudenko committed:
style: Fix radio customization

OleksiyRudenko committed:
general: Add href for telephone. Customize <a> style

OleksiyRudenko committed:
style: Introduce css-reset

OleksiyRudenko committed:
style: Add better radio styling

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
general: Add image assets

OleksiyRudenko committed:
html: Add icons to links

OleksiyRudenko committed:
style: Change layout and style for source selection

OleksiyRudenko committed:
style: File upload control

OleksiyRudenko committed:
style: Fix radio buttons imagery styling

OleksiyRudenko committed:
style: Restyle tone of voice radio buttons set

OleksiyRudenko committed:
style: Comment out obsolete style

OleksiyRudenko committed:
style: Restyle range input control

OleksiyRudenko committed:
html: Add standard range input control labelling

OleksiyRudenko committed:
style: Improve radio buttons styling

OleksiyRudenko committed:
style: Form action buttons

OleksiyRudenko committed:
style: Update text color

OleksiyRudenko committed:
style: Stylize select drop down control

OleksiyRudenko committed:
html: Add titles to Remove buttons

OleksiyRudenko committed:
style: Main section radius

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: Footer flex

OleksiyRudenko committed:
style: Header

OleksiyRudenko committed:
style: Add normalize

OleksiyRudenko committed:
style: Header

OleksiyRudenko committed:
style: Radio tone of voice

OleksiyRudenko committed:
general: Add flex element template

OleksiyRudenko committed:
style: Update header styles

OleksiyRudenko committed:
general: Add flex element template

OleksiyRudenko committed:
style: Update header styles

OleksiyRudenko committed:
style: Range input

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: Improve header styling

OleksiyRudenko committed:
style: Improve order header styling

OleksiyRudenko committed:
Merge remote-tracking branch 'origin/master'

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: Document type

OleksiyRudenko committed:
style: Sections counter

OleksiyRudenko committed:
style: Misc

OleksiyRudenko committed:
style: Misc

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: Migrate to em measurements

OleksiyRudenko committed:
Update comments

OleksiyRudenko committed:
style: Footer

OleksiyRudenko committed:
style: Urgency section

OleksiyRudenko committed:
style: Languages

OleksiyRudenko committed:
style: Form footer section

OleksiyRudenko committed:
style: Resize fonts

OleksiyRudenko committed:
style: Restyle Sign-Up section

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: Fix footer blocks styling

OleksiyRudenko committed:
style: Fix css counter styling

OleksiyRudenko committed:
style: Fix styling for custom control

OleksiyRudenko committed:
style: Fix payment method radios style

OleksiyRudenko committed:
style: Minor font-size fix

OleksiyRudenko committed:
style: Fix footer block width style

OleksiyRudenko committed:
style: Make header more flexible

OleksiyRudenko committed:
style: Make order header more flexible

OleksiyRudenko committed:
style: Remove height for <a>

OleksiyRudenko committed:
style: Set min-height for payment method radio control

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: Fix FA font-weight to display icons properly

OleksiyRudenko committed:
style: Adjust validity markers positioning

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
style: Stylize text input placeholder

OleksiyRudenko committed:
html: Make non-submission and non-resetting buttons explicitly buttons

OleksiyRudenko committed:
bugfix: accepted extensions

OleksiyRudenko committed:
bugfix: Add more MS Word like extensions

OleksiyRudenko committed:
bugfix: Make radio items divs

OleksiyRudenko committed:
bugfix: Set valid min and max attributes for input type=month

OleksiyRudenko committed:
bugfix: Remove value attribute from td element since not allowed

OleksiyRudenko committed:
Update time tracking record

OleksiyRudenko committed:
docs: Add refs to HTML and CSS validators

OleksiyRudenko committed:
bugfix: Make TnC checkbox unchecked and required

OleksiyRudenko committed:
docs: Add snippets section

OleksiyRudenko committed:
style: Fix upper form buttons styling

OleksiyRudenko committed:
Move codebase to task_11 subdirectory

OleksiyRudenko committed:
Import html5form repo

OleksiyRudenko committed:
chores: Remove obsolete commented out declarations

OleksiyRudenko committed:
style: Highlight hovered buttons

OleksiyRudenko committed:
style: Highlight hovered radio buttons

OleksiyRudenko committed:
style: Highlight hovered select

OleksiyRudenko committed:
style: Highlight hovered file upload control

AMashoshyna commented:
@OleksiyRudenko Great work! 
Possible improvements:
1. Check markup validation errors: https://validator.w3.org/nu/?doc=https%3A%2F%2Foleksiyrudenko.github.io%2Fassignments_OleksiyRudenko%2Ftask_11%2Findex.html
2. Make radio buttons accessible from keyboard (switch between options with arrow keys), access them with tab key

Andrew-Omelchenko commented:
Great job! Some remarks:
1. Text inside input fields is black. It may have the #6e768f color.
2. CHOOSE A PASSWORD field always remains invalid.
3. CODE field lets type more than 3 digits and stays valid. (Try to add the maxlength="3" attribute).
4. Input in CODE field is always visible. Try to hide it (type="password").

AMashoshyna commented:
![image](https://user-images.githubusercontent.com/17703790/35613163-ecaeec00-0673-11e8-8ca1-3d9fcc0c9831.png)
this control misses on-focus style

Masmik commented:
<img width="487" alt="screen shot 2018-01-31 at 10 49 32" src="https://user-images.githubusercontent.com/24464487/35613361-86aa4a02-0674-11e8-905e-ec3560f374a7.png">

Трішки позлітали сірі кола в input[type=range] і підпис avarage добре було б зробити жовтого кольору

lempiy commented:
OMG so many commits ! :-)
squash them upon merge please .))

Masmik commented:
<img width="755" alt="screen shot 2018-01-31 at 11 37 17" src="https://user-images.githubusercontent.com/24464487/35615558-228aa9c0-067b-11e8-96a8-8959af9849ae.png">
there is no border on the table

OleksiyRudenko committed:
Remove irrelevant document

OleksiyRudenko committed:
Update links to live demo and html5 validation service

OleksiyRudenko committed:
bugfix: Input text color

OleksiyRudenko committed:
bugfix: Set CVV code length to 3 and assign type password

OleksiyRudenko committed:
style: Add styling for uploaded file listing

OleksiyRudenko committed:
html: Expand on AmEx cards

OleksiyRudenko commented:
@Andrew-Omelchenko thank you for pointing out the issues. Items 1, 3 and 4 fixed.
Item 2: couldn't reproduce. Could you, please, check once again? Please, note that validation tooltip and on-hover hint says "Must contain at least one number and one uppercase and lowercase letter, and at least 8 characters (32 max)"
[Live demo](https://kottans-frontend-2018.github.io/assignments_OleksiyRudenko/task_11/index.html)

OleksiyRudenko committed:
html: Expand on AmEx cards

OleksiyRudenko commented:
@AMashoshyna , @Masmik , thank you for your review! Will fix. Added to [issues](https://github.com/kottans-frontend-2018/assignments_OleksiyRudenko/issues)
@lempiy , [will do](https://github.com/kottans-frontend-2018/assignments_OleksiyRudenko/issues/7) :)

OleksiyRudenko committed:
bugfix: Add alt attribute to img elements

Andrew-Omelchenko reviewed (approved):
Agree with you. Item 2 works well.

OleksiyRudenko committed:
bugfix: Comment out inputmode attribute to pass html5 strict validation

OleksiyRudenko committed:
bugfix: Remove required attr as it is not allowed for type=range

OleksiyRudenko committed:
bugfix: Replace value attr with title attr

OleksiyRudenko committed:
bugfix: Replace input type=month with type=date to pass html5 strict validation

JS polyfill not employed as major project target is html5 valid code rather than strictest adherence to business logic

OleksiyRudenko committed:
bugfix(reversal): Reverted to input type=month.

Input type=date still produces an html5 validation warning.

OleksiyRudenko committed:
bugfix: Employ data-autocomplete

OleksiyRudenko committed:
bugfix: Replace section with div for radio groups

Context is irrelevant for section employment

OleksiyRudenko committed:
bugfix: Make styled radio buttons layout html5 compliant

OleksiyRudenko committed:
bugfix: Make file upload button focusable

OleksiyRudenko committed:
style: Add style for range input labels

OleksiyRudenko committed:
bugfix: Add a control-focus element after input to make it selectable in css

OleksiyRudenko committed:
bugfix: Make button a label for control

OleksiyRudenko committed:
bugfix: Make radio buttons keyboard friendly

OleksiyRudenko committed:
bugfix: Make file upload control visually focusable, hoverable, kbd-selectable

Resolves #3

OleksiyRudenko commented:
@AMashoshyna 

 1. Radio buttons not navigable with kbd - `Fixed`. Please, check.

 2. File upload control - `Fixed`. Please, check.

 3. HTML5 validation - almost `Fixed`

The latter fires a warning:
> The `month` input type is not supported in all browsers. Please be sure to test, and consider using a polyfill.

Type `date` fires similar warning. Polyfill requires JS, to my knowledge.

Can we, please, tolerate the warning re `month`?

* [Demo](https://kottans-frontend-2018.github.io/assignments_OleksiyRudenko/task_11/index.html)
* [HTML5 validator](https://validator.w3.org/nu/?doc=https%3A%2F%2Fkottans-frontend-2018.github.io%2Fassignments_OleksiyRudenko%2Ftask_11%2Findex.html)
* [repo](https://github.com/kottans-frontend-2018/assignments_OleksiyRudenko/tree/task11/html5form/task_11)

OleksiyRudenko commented:
@Masmik ,
 1. 'Average' accented - `Fixed`.

 2. Misplaced edge markers:

Couldn't reproduce. It also might be happening on an older version of the page as the link in the top post has been a bit outdated. Can you please [check it once again](https://kottans-frontend-2018.github.io/assignments_OleksiyRudenko/task_11/index.html)?

If there are still any issues a screenshot with dev tools open will be much appreciated.
![screenshot_155](https://user-images.githubusercontent.com/8686631/35722092-94c662bc-07fd-11e8-921b-d18695aeb06b.png)

Masmik commented:
I still have a problem in this block (Chrome window.innerWidth = 1920px MacOs)
![screen shot 2018-02-02 at 9 46 16 am](https://user-images.githubusercontent.com/24464487/35722207-0c33d5a0-07fe-11e8-9c38-2975b7ea2230.png)

solves this problem in dev tools:
input[type=range].urgency:before{
margin-top: 0.4rem;
}
input[type=range].urgency:after{
margin-top: 0.4rem;
}

OleksiyRudenko committed:
bugfix: Adjust range edge markers positioning

OleksiyRudenko commented:
@Masmik , thank you for your suggestion. Well, being applied makes it looking bad under Chrome for Windows.
Instead, did the following:
```
top: 50%;
margin-top: -0.2rem;
```
and `height: 3rem;` for its container, hoping it will look same on all browsers.

Can I know your platform and browser version?

Masmik commented:
working!!!
really good idea
